js/fcoin.js

```diff
@@ -0,0 +1,572 @@
+        let response = await this.privateGetOrdersOrderId (this.extend ({
+            'order_id': id,
+        }, params));
+        return this.parseOrder (response);
+    }
+
+    async fetchOpenOrders (symbol = undefined, since = undefined, limit = undefined, params = {}) {
+        let result = undefined;
+        result = await this.fetchOrders (symbol, since, limit, { 'states': 'submitted' });
+        return result;
+    }
+
+    async fetchClosedOrders (symbol = undefined, since = undefined, limit = undefined, params = {}) {
+        let result = undefined;
+        result = await this.fetchOrders (symbol, since, limit, { 'states': 'filled' });
+        return result;
+    }
+
+    async fetchOrders (symbol = undefined, since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        let market = this.market (symbol);
+        let response = await this.privateGetOrders (this.extend ({
+            'symbol': market.id,
+            'states': 'submitted',
+        }, params));
+        return this.parseOrders (response['data']);
+    }
+
+    parseOHLCV (ohlcv, market = undefined, timeframe = '1m', since = undefined, limit = undefined) {
+        return [
+            ohlcv['seq'],
+            ohlcv['open'],
+            ohlcv['high'],
+            ohlcv['low'],
+            ohlcv['close'],
+            ohlcv['base_vol'],
+        ];
+    }
+
+    async fetchOHLCV (symbol, timeframe = '1m', since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        if (typeof limit === 'undefined') {
+            limit = 100;
+        }
+        let market = this.market (symbol);
+        let request = {
+            'symbol': market['id'],
+            'timeframe': this.timeframes[timeframe],
+            'limit': limit,
+        };
+        let response = await this.marketGetCandlesTimeframeSymbol (this.extend (request, params));
+        return this.parseOHLCVs (response, market, timeframe, since, limit);
+    }
+
+    nonce () {
+        return this.milliseconds ();
+    }
+
+    sign (path, api = 'public', method = 'GET', params = {}, headers = undefined, body = undefined) {
+        let request = '/' + this.implodeParams (path, params);
+        let query = this.omit (params, this.extractParams (path));
+        let url = this.urls['api'];
+        if ((api === 'public') || (api === 'market') || (path.indexOf ('/hist') >= 0)) {
+            request = '/' + this.version + '/' + api + request;
+            url += request;
+            if (Object.keys (query).length) {
+                url += '?' + this.urlencode (query);
+            }
+        }
+        if (api === 'private') {
+            this.checkRequiredCredentials ();
+            body = this.json (query);
+            request = '/' + this.version + request;
+            url += request;
+            let paramsStr = '';
+            let keys = Object.keys (query);
+            keys = keys.sort ();
```

Comment:
Unfortunately, this line won't transpile properly, we usually use `this.keysort` (ctrl-f and see other exchanges for `keysort`).